cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -127,7 +128,11 @@ func GetNodeRegistration(kubeconfigFile string, client clientset.Interface, node
 	// gets the name of the current node
 	nodeName, err := getNodeNameFromKubeletConfig(kubeconfigFile)
 	if err != nil {
-		return errors.Wrap(err, "failed to get node name from kubelet config")
+		klog.Warningf("failed to get node name from %s: %v, falling back to local hostname", kubeconfigFile, err)
+		nodeName, err = nodeutil.GetHostname("")
+		if err != nil {
+			return err
+		}
```

Comment:
this is a bit questionable, but the fallback is not so bad, i think.
it allows the dryrun without a kubelet.conf, at the same time the warning makes it obvious to the user that something is wrong and they would face a problem if hostname != nodename outside of dryrun.